tests/unit/src/aws_iot_tests_unit_json_utils_helper.c

```diff
@@ -58,26 +60,48 @@ TEST_C(JsonUtils, ParseStringBasic) {
 TEST_C(JsonUtils, ParseStringLongerStringIsValid) {
 	int r;
 	const char *json = "{\"x\":\"this is a longer string for test 2\"}";
-	char parsedString[50];
+	char parsedString[STRING_BUFFER_LENGTH];
 
 	IOT_DEBUG("\n-->Running Json Utils Tests - Parse long string \n");
 
 	r = jsmn_parse(&test_parser, json, strlen(json), t, sizeof(t) / sizeof(t[0]));
-	rc = parseStringValue(parsedString, json, t + 2);
+	rc = parseStringValue(parsedString, STRING_BUFFER_LENGTH, json, t + 2);
 
 	CHECK_EQUAL_C_INT(3, r);
 	CHECK_EQUAL_C_INT(SUCCESS, rc);
 	CHECK_EQUAL_C_STRING("this is a longer string for test 2", parsedString);
 }
 
+/* Test that parsing a string doesn't overflow the given buffer. */
+TEST_C(JsonUtils, ParseStringWithBufferTooSmall) {
+	static const char* const pJsonString = "{\"key\":\"This value is longer than JSON_BUFFER_LENGTH, which should be 50.\"}";
+	char parsedString[STRING_BUFFER_LENGTH] = {0};
+	int jsmnReturn, i;
+
+	IOT_DEBUG("\n-->Running Json Utils Tests -  String parsing with buffer too small. \n");
+
+	jsmnReturn = jsmn_parse(&test_parser, pJsonString, strlen(pJsonString), t, sizeof(t)/sizeof(t[0]));
+	CHECK_EQUAL_C_INT(3, jsmnReturn);
+
+	rc = parseStringValue(parsedString, STRING_BUFFER_LENGTH, pJsonString, t + 2);
+
+	CHECK_EQUAL_C_INT(SHADOW_JSON_ERROR, rc);
+
+	/* Ensure there was no attempt to write to a buffer that's too small. */
+	for(i = 0; i < STRING_BUFFER_LENGTH; i++)
```

Comment:
NP: You could do a mem compare

Reply:
Wouldn't that require keeping an additional block of all-zero memory around?